From bebcdd35103876a6b2992653950d93372f33c1db Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Wed, 29 Jan 2020 17:10:22 +0100
Subject: [PATCH] pxe: First try for fdtfile env variable

If fdtfile is set in env, from resinOS_uEnv.txt for instance,
then it will be used to load the dtb.

It's path should be indentical to an FDT entry from extlinux.conf,
i.e: /boot/mycustom.dtb, which should be relative to
/mnt/sysroot/active/current/boot/.

Setting FDT in extlinux.conf will override env's fdtfile.
Not choosing any will lead to default behaviour, where the
DTB read from kernel_dtb partition by cboot gets passed to
u-boot and then to the kernel.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 cmd/pxe.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmd/pxe.c b/cmd/pxe.c
index 7de0559e42..542dacaa5d 100644
--- a/cmd/pxe.c
+++ b/cmd/pxe.c
@@ -742,6 +742,9 @@ static int label_boot(cmd_tbl_t *cmdtp, struct pxe_label *label)
 		char *fdtfile = NULL;
 		char *fdtfilefree = NULL;
 
+		/* If label has no 'FDT', get dtb from env, if exists */
+		fdtfile = getenv("fdtfile");
+
 		if (label->fdt) {
 			fdtfile = label->fdt;
 		} else if (label->fdtdir) {
-- 
2.17.1

